File: exercise/php/day12/src/Gifts/Santa.php
<?php

namespace Gifts;

use InvalidArgumentException;

class Santa
{
    private array $childrenRepository = [];

    public function chooseToyForChild(string $childName): ?Toy
    {
        $found = null;
        foreach ($this->childrenRepository as $currentChild) {
            if ($currentChild->name === $childName) {
                $found = $currentChild;
                break;
            }
        }

        if ($found === null) {
            throw new InvalidArgumentException("No such child found");
        }

        switch ($found->behavior) {
            case "naughty":
                return $found->wishlist[2];
            case "nice":
                return $found->wishlist[1];
            case "very nice":
                return $found->wishlist[0];
            default:
                return null;
        }
    }

    public function addChild(Child $child): void
    {
        $this->childrenRepository[] = $child;
    }
}

File: exercise/C#/day12/Gifts/Santa.cs
namespace Gifts;

public class Santa
{
    private readonly List<Child> _childrenRepository = [];

    public Toy? ChooseToyForChild(string childName)
    {
        Child? found = null;
        for (int i = 0; i < _childrenRepository.Count; i++)
        {
            var currentChild = _childrenRepository[i];
            if (currentChild.Name == childName)
            {
                found = currentChild;
            }
        }

        if (found == null)
            throw new InvalidOperationException("No such child found");

        if (found.Behavior == "naughty")
            return found.Wishlist[^1];

        if (found.Behavior == "nice")
            return found.Wishlist[1];

        if (found.Behavior == "very nice")
            return found.Wishlist[0];

        return null;
    }

    public void AddChild(Child child) => _childrenRepository.Add(child);
}